cmd/router/main_test.go

```diff
@@ -515,3 +515,16 @@ func TestCallServiceWhenMultipleHeadersToPropagate(t *testing.T) {
 	fmt.Printf("final response:%v\n", response)
 	assert.Equal(t, expectedResponse, response)
 }
+
+
+func TestMalformedURL(t *testing.T) {
+	
+        malformedURL := "http://single-1.default.{$your-domain}/switch"
+        _, _, err := callService(malformedURL, []byte{}, http.Header{})
+        if err == nil {
```

Comment:
Okay, I've added it.

Reply:
According to the above logic the err should not be nil right?

Reply:
I do not think the testing code is executed

Reply:
@yuzisun My initial thought was let the test failed by t.fatal because the call service didn't handle error correctly even though err == nil. But I think your logic is more correct, so I made the  corrections.